docs/technical-details/reference/graphql.md

```diff
@@ -0,0 +1,235 @@
+# GraphQL Querying
+
+HarperDB supports GraphQL in a variety of ways. It can be used for [defining schemas](../../developers/applications/defining-schemas.md), and for querying [Resources](./resource.md).
+
+Get started by setting `graphql: true` in `config.yaml`.
+
+This automatically enables a `/graphql` endpoint that can be used for GraphQL queries.
+
+> HarperDB's GraphQL component is inspired by the [GraphQL Over HTTP](https://graphql.github.io/graphql-over-http/draft/#) specification; however, it does not fully implement neither that specification nor the [GraphQL](https://spec.graphql.org/) specification.
+
+Queries can either be `GET` or `POST` requests, and both follow essentially the same request format. `GET` requests must use search parameters, and `POST` requests use the request body.
+
+For example, to request the GraphQL Query:
+```graphql
+query GetDogs {
+	Dog {
+		id
+		name
+	}
+}
+```
+The `GET` request would look like:
+```sh
+curl 'http://localhost:9926/graphql?query=query+GetDogs+%7B+Dog+%7B+id+name+%7D+%7D'
+```
+And the `POST` request would look like:
+```sh
+curl 'http://localhost:9926/graphql' \
+	-H 'Content-Type: application/json' \
+	-d '{ "query": "query GetDogs { Dog { id name } } }" '
+```
+
+The HarperDB GraphQL querying system is strictly limited to HarperDB Resources. Queries can only specify HarperDB Resources and their attributes in the selection set. Queries can filter using [arguments](https://graphql.org/learn/queries/#arguments) on the top-level Resource field. HarperDB provides a short form pattern for simple queries, and a long form pattern based off of the [Resource Query API](./resource.md#query) for more complex queries.
```

Comment:
Is `resources` the best word here? Applications/components? or maybe say any table that is defined with a graphql schema?

Reply:
"Resource" is technically a correct definition of what can be queried here. However, it is more specifically the _exported_ resources  (a table is one type of resource, but other resources can be exported). And it might be helpful to include a note or parenthetical that this includes `@export`'ed tables from the schema or resources.js.

Reply:
out of curiosity, how would you query a resource that isn't a table resource?

Reply:
I've included a new line here that may provide some additional clarity.